ocs/cluster.py

```diff
@@ -0,0 +1,248 @@
+        _url_prefix (str): url prefix for REST api interaction
+    """
+
+    def __init__(self, **config):
+        # cluster_name is name of cluster in rook of type CephCluster
+        self._cluster_name = config.get('cluster_name', 'rook-ceph')
+        """
+        TODO: Instantiate CephCluster object somewhere higher after deployment
+        Read namespace and cluster name from config.
+        """
+        self._namespace = config.get(
+            'namespace', ENV_DATA['cluster_namespace']
+        )
+        self._ocs_pods = pod.get_all_pods(self._namespace)
+
+        #self.cluster_resource = os.path.join(
+        #    default.TEMPLATE_DIR, "cluster.yaml")
+        # Once we figure out where to initialize this CephCluster object we
+        # will pass **config
+        #self.cluster_config = generate_yaml_from_jinja2_template_with_data(
+        #    self.cluster_resource, **config
+        #)
+        self.cluster_config = CEPHCLUSTER.get()['items'][0]
+        self.cephfs_config = default.CEPHFILESYSTEM_DICT.copy()
+        #Below is just for testing purpose
+        #self.cephfs_config['metadata']['name'] = 'cephfs-test-0913235459'
+
+        # We are not invoking ocs.create() here
+        # assuming cluster creation is done somewhere after deployment
+        # So just load ocs with existing cluster details
+        self.cluster = ocs.OCS(**self.cluster_config)
+        self.cephfs = ocs.OCS(**self.cephfs_config)
+
+        self.cluster.reload()
+        self.cephfs.reload()
+
+        self.mon_selector = default.MON_APP_LABEL
+        self.mds_selector = default.MDS_APP_LABEL
+        self.tool_selector = default.TOOL_APP_LABEL
+
+        self.mons = []
+        self.mdss = []
+        self.toolbox = []
+
+        self.scan_cluster()
+        self.mon_count = len(self.mons)
+        self.mds_count = len(self.mdss)
+
+    @property
+    def cluster_name(self):
+        return self._cluster_name
+
+    @property
+    def namespace(self):
+        return self._namespace
+
+    @property
+    def pods(self):
+        return self._ocs_pods
+
+    def scan_cluster(self):
+        """
+        Get accurate info on current state of pods
+        """
+        self._ocs_pods = pod.get_all_pods(self._namespace)
+        self.mons = self._filter_pods(self.mon_selector, self.pods)
+        self.mdss = self._filter_pods(self.mds_selector, self.pods)
+        self.toolbox = self._filter_pods(self.tool_selector, self.pods)
+
+        # set port attrib on mon pods
+        self.mons = list(map(self.set_port, self.mons))
+
+    @staticmethod
+    def set_port(pod):
+        """
+        Set port attribute on pod
+        Args:
+            pod(Pod): Pod object
+        """
+        l1 = pod.pod_data.get('spec').get('containers')
+        l2 = l1[0]['ports'][0]['containerPort']
+        # Dynamically added attribute 'port'
+        pod.port = l2
+        logging.info(f"port={pod.port}")
+        return pod
```

Comment:
why this function returns a pod and not bool.
Looks strange

Reply:
At line 112 I am using map function, which returns a new list with whatever returned by set_port command. Hence returning pod.

Reply:
Sorry for bringing this back again.
you have the pod as you pass it as a parameter to the function.
Why returning it back and not something that will let you know if the operation, set_port, succeeds or not